index.js

```diff
@@ -93,25 +93,59 @@ function parseListItem(listItem) {
   entry.url = link.url;
   entry.title = link.children[0].value;
   // remember to get OTHER STUFF!! remember there may be multiple links!
+  let insideAuthors = false; // are we still parsing authors across AST nodes?
   for (let i of otherStuff) {
     if (s === "") {
       // this is almost always, except for when we are parsing a multi-element note
-      if (i.type === "text" && i.value.slice(0, 3) === " - ") {
-        // author found
-        let parenIndex = i.value.indexOf("(");
-        if (parenIndex === -1) {
-          entry.author = i.value.slice(3).trim();
-        } else {
-          entry.author = i.value.slice(3, parenIndex).trim(); // go from " - " until the first "("
+      if (i.type === "text") {
+        const text = i.value;
+        const parenIndex = text.indexOf("(");
+
+        if (insideAuthors) {
+          // an author with role entity found. (maybe after some inlineCode node)
+          // so, append until next note token, if any
+          entry.author +=
+            parenIndex === -1 ? text : text.substring(0, parenIndex);
+        }
+
+        if (text.startsWith(" - ")) {
+          // authors found
+          insideAuthors = true;
+          entry.author =
+            parenIndex === -1
+              ? text.slice(3) // go from " - " until the last char
+              : text.slice(3, parenIndex); // go from " - " until the first "("
+        }
+
+        if (parenIndex !== -1) {
+          // notes found (currently assumes no nested parentheses)
+          insideAuthors = false;
+          if (entry.notes === undefined) entry.notes = [];
+          leftParen = parenIndex;
+          while (leftParen != -1) {
+            rightParen = text.indexOf(")", leftParen);
+            if (rightParen === -1) {
+              // there must be some *emphasis* found
+              s += text.slice(leftParen);
+              break;
+            }
+            entry.notes.push(text.slice(leftParen + 1, rightParen));
+            leftParen = text.indexOf("(", rightParen);
+          }
         }
       }
+      if (insideAuthors && i.type === "inlineCode") {
+        // author role found. append rebuilding markdown format and then move on
+        entry.author += "`" + i.value + "`";
```

Comment:
seems like the values accepted as relators should be limited!

Reply:
Are you talking about limit to `trl.:` ? I think this is more to take into account reviewing PRs or in the linter

Reply:
for the parser, mostly don't want it to be tripped up by syntax we haven't anticipated.

Reply:
I have not fully understood what you meant by the last two posts. Could you develop the premise better or tell me how I should code it?